source/change-streams/change-streams.rst

```diff
@@ -388,26 +388,26 @@ Single Server Topologies
 
 Presently, change streams cannot be initiated on single server topologies as they do not have an oplog.  Drivers MUST NOT throw an exception in this scenario, but instead rely on an error returned from the server.  This allows for the server to seamlessly introduce support for this in the future, without need to make changes in driver code.
 
-startAtClusterTime
+startAtOperationTime
 ^^^^^^^^^^^^^^^^^^
 
 :since: 4.0
 
-``startAtClusterTime`` specifies that a ``changeStream`` will only return changes that occurred at or after the specified ``Timestamp``.
+``startAtOperationTime`` specifies that a change stream will only return changes that occurred at or after the specified ``Timestamp``.
 
-The server expects ``startAtClusterTime`` as a bson document of the form
+The server expects ``startAtOperationTime`` as a BSON Timestamp
 
 .. code:: typescript
 
     {
       ts: Timestamp;
     }
 
-Drivers MUST allow users to specify a ``startAtClusterTime`` option in the ``Collection.watch`` and ``Database.watch`` helpers. They MUST allow users to specify this value as a raw ``Timestamp``, and MUST NOT force users to wrap the timestamp as the server specifies.
+Drivers MUST allow users to specify a ``startAtOperationTime`` option in the ``Collection.watch`` and ``Database.watch`` helpers. They MUST allow users to specify this value as a raw ``Timestamp``.
 
-``startAtClusterTime`` and ``resumeAfter`` are mutually exclusive; if both ``startAtClusterTime`` and ``resumeAfter`` are set, the server will return an error. Drivers MUST NOT throw a custom error, and MUST defer to the server error.
+``startAtOperationTime`` and ``resumeAfter`` are mutually exclusive; if both ``startAtOperationTime`` and ``resumeAfter`` are set, the server will return an error. Drivers MUST NOT throw a custom error, and MUST defer to the server error.
 
-If neither ``startAtClusterTime`` or ``resumeAfter`` are specified, and the server version is >= ``4.0`` drivers MUST set a default ``startAtClusterTime`` value using a ``$clusterTime`` from any server response (ex: ``ismaster``). This allows change streams to be resumed before the first notification is received.
+If neither ``startAtOperationTime`` nor ``resumeAfter`` are specified, and the server version is >= ``4.0`` drivers MUST set a default ``startAtOperationTime`` (a default timestamp can be obtained from the ``operationTime`` field on any server response, like an ``isMaster``. This allows change streams to be resumed before the first notification is received.
```

Comment:
The parenthetical is missing a closing parenthesis. I would suggest writing it out as a normal sentence, though (perhaps with "(e.g. `isMaster`)" at the end).

This still seems a bit ambiguous. In https://github.com/mongodb/specifications/pull/326#discussion_r190962952 you wrote:

> The server spec for startAtOperationTime actually states that the operation time must be obtained as early as possible, rather than on the fly.

Wouldn't we want to use the most recently observed operation time? If we merely use the earliest collected operation time, there's no telling how old that may be if the topology has been persisted. Tracking of operation time may fall outside the scope of this spec, so this may be a question for @ajdavis and/or @rstam.

Reply:
I actually misspoke. The operation time should be the most recent operation time when the changeStream is created. That operationTime should be tracked by the changeStream until it receives its first resume token.

Reply:
If I understand correctly, the initial `aggregate` command should omit `startAtOperationTime` if not provided by the user. It should then capture `startAtOperationTime` from the `aggregate` response and use that for any resume behavior (if the change stream must resume before an actual change comes back with a proper resume token). If so, let's just make sure this is clear in the spec (if it isn't already).

Reply:
No, the only reason to omit a `startAtOperationTime` against 4.0 is if there is a resume token. Any changeStream without a resume token should include `startAtOperationTime`, either one provided by the user, or one based on a previous command like an `isMaster`.